### PR TITLE
refactor(send): Add context.Context to parameter list of Send method

### LINKED ISSUE #49: refactor(send): Add context.Context to notify.Send method
**Is your feature request related to a problem? Please describe.**

Since _notify.Send_ may basically send a message to a huge amount of receivers, the user might appreciate to have the ability to cancel long running send executions.

**Describe the solution you'd like**

To achieve this behavior we will add a _context.Context_ to the parameters of _notify.Send_.

Current _notify.Send_ signature:

```go
func (n Notify) Send(subject, message string) error {
    ...
}
```

New _notify.Send_ signature:

```go
func (n Notify) Send(ctx context.Context, subject, message string) error {
    ...
}
```

This needs to be done for every services implementation of the send method. The respective implementations will then need to check if the context got cancelled.

Example implementation:

```go
func (s SomeService) Send(ctx context.Context, subject, message string) error {
    
    ...

    for _, receiver := range s.receivers {
        select {
	    case <-ctx.Done():
	        return nil // Context got cancelled; just return
            default:
	        _, _ = t.client.Send(receiver, message)  
	    }
        }
    }	

    return nil
}
```

### PR DESCRIPTION
**Pull request by nikoksr**
## Info

This closes #49. I added _context.Context_ to the parameter list to all implementations and definitions of the _Send_ method. This will allow users to cancel long running calls of said function.

## Suggested changes

- Add _context.Context_ to parameter list of _Send_ method
- Update tests
- Update documentation

## Additional context

~~I need feedback for this! I added some _TODO_ comments here and there.~~

In some cases the external library does not accept a _context.Context_ itself and we don't loop over receivers but pass them all at once. This makes it hard for me to decide how to handle the context correctly. Do we check  _context.Context.IsDone_ at the beginning of a _Send_ implementation, do we check it right before actually sending the message payload to the external service?

### CODE REVIEW

**Comment by nikoksr** [comment]
@checkaayush could you give this a quick review?

File: service/whatsapp/whatsapp.go [hunk 1]
```diff
@@ -19,7 +20,7 @@ const (
 
 var sessionFilePath = filepath.Join(os.TempDir(), "whatsappSession.gob")
 
-// whatsappClient abstracts go-whatsapp for writing unit tests
+// mockWhatsappClient abstracts go-whatsapp for writing unit tests
```
Comment (checkaayush) [inline service/whatsapp/whatsapp.go:23]: Should be `whatsappClient` instead of `mockWhatsappClient `.

**Review by checkaayush — APPROVED** [review]
LGTM. Just a minor typo fix suggested.

File: service/whatsapp/whatsapp.go (hunk 1 above)
Comment (nikoksr) [reply service/whatsapp/whatsapp.go]: Appreciate it! Fixed this for _Plivo_ service too.

**Pull request merged**